packages/lib/src/components/Klarna/components/KlarnaWidget/KlarnaWidget.tsx

```diff
@@ -76,21 +77,30 @@ export function KlarnaWidget({ sdkData, paymentMethodType, payButton, ...props }
         } catch (e) {
             handleError();
         }
-    };
+    }, [sdkData.payment_method_category, props.onComplete, props.onError]);
+
+    /**
+     * Initializes Klarna SDK if it is already available and reinitialize
+     * it when the init time refreshes
+     */
+    useEffect(() => {
+        const isKlarnaAvailable = window.Klarna?.Payments?.init;
+        if (isKlarnaAvailable) {
+            initializeKlarnaWidget();
+        }
+    }, [widgetInitializationTime]);
 
-    // Add Klarna Payments Widget SDK
     useEffect(() => {
         window.klarnaAsyncCallback = function () {
             initializeKlarnaWidget();
         };
-
         const script = new Script(KLARNA_WIDGET_URL);
```

Comment:
You only need to load the script once, for the first widget.
After that everything works now that the widget is reinitialised.
So you could having something like this here:
```
        const isKlarnaAvailable = window.Klarna?.Payments?.init;
        if (!isKlarnaAvailable) {
            const script = new Script(KLARNA_WIDGET_URL);
            void script.load();

            return () => {
                script.remove();
            };
        }
```